### ISSUE TITLE
WSL2 - Docker can't build PHP container

### ISSUE THREAD

**Issue by eyeskiller** [Sylius/Sylius-Standard#454]
Hi, I am trying to install Sylius via Docker on WSL2 and I am receiving following error while building PHP container

`ERROR: Service 'php' failed to build : COPY failed: stat /var/lib/docker/tmp/docker-builder139509019/composer.lock: no such file or directory`

**Comment by achref-riahi** [comment]
After the clone of repo, you should run composer install to generate this file (composer.lock) and to download the project Dependencies, But Note, you should run composer with php version 7.3. For example
`/usr/bin/php7.3 /usr/bin/composer install` 
To know your php versions path and the composer path on Linux you can use:
`whereis composer` and `whereis php`

good luck!

**Comment by aHekuaH** [comment]
But why isn't there a composer.lock to begin with? Shouldn't it be part of this repo? In older Sylius Versions it used to be there

**Comment by Cool-and-Deadly** [comment]
Am having the same problem. I would have expected to run composer install from inside the container. Is it true that this should be done from outside?

**Comment by eiannone** [comment]
I'm getting the same error. This is due to these lines in Dockerfile:
```
# prevent the reinstallation of vendors at every changes in the source code
COPY composer.json composer.lock symfony.lock ./
RUN set -eux; \
	composer install --prefer-dist --no-autoloader --no-scripts --no-progress --no-suggest; \
	composer clear-cache
```

Before running composer install, it is expecting that composer.lock file is already there, so it should be included in the repository. Currently this file is excluded through .gitignore, but there is a pull request (https://github.com/Sylius/Sylius-Standard/pull/505) to include it again, as requested in issue https://github.com/Sylius/Sylius-Standard/issues/503.

Running composer install, as proposed by @achref-riahi , produces this warning message:
`No lock file found. Updating dependencies instead of installing from lock file. Use composer update over composer install if you do not have a lock file.`

So in this way it does an `update `instead of an `install`.

Anyway, if you exclude composer.lock from copying, Docker will build the image succesfully.

**Comment by cristi-contiu** [comment]
This error also appears when following the install instructions provided here: https://sylius.com/download/ in the "How to install with Docker" section.

So in order to build the image you need to run `composer install`, but you may not have PHP & Composer installed locally and rely on the Docker image to provide them.

Maybe there can be some kind of flag, arg or env var that allows building the image without a `composer.lock` file?

**Comment by tarnawski** [comment]
The proposition of fix in https://github.com/Sylius/Sylius-Standard/pull/559

**Comment by lchrusciel** [comment]
Should be fixed on master in #536
